src/scenes/Wallet/components/History/index.js

```diff
@@ -11,96 +12,79 @@ import {
-import { TransactionType, SortBy } from '../../../../constants';
+import { SortBy } from '../../../../constants';
 import Config from '../../../../config/app';
 import { getShortLocalDateTimeString } from '../../../../helpers/utility';
-import graphqlActions from '../../../../redux/Graphql/actions';
-
 
 @injectIntl
 @withStyles(styles, { withTheme: true })
 @connect((state) => ({
   syncBlockNum: state.App.get('syncBlockNum'),
-  transactions: state.Graphql.get('getTransactionsReturn'),
   txReturn: state.Graphql.get('txReturn'),
-}), (dispatch) => ({
-  getTransactions: (filters, orderBy, limit, skip) => dispatch(graphqlActions.getTransactions(filters, orderBy, limit, skip)),
 }))
+@inject('store')
+@observer
 export default class WalletHistory extends Component {
   static propTypes = {
     classes: PropTypes.object.isRequired,
-    getTransactions: PropTypes.func.isRequired,
-    transactions: PropTypes.array,
     txReturn: PropTypes.object,
     syncBlockNum: PropTypes.number.isRequired,
   };
 
   static defaultProps = {
-    transactions: [],
     txReturn: undefined,
   };
 
-  state = {
-    transactions: [],
-    order: SortBy.Descending.toLowerCase(),
-    orderBy: 'createdTime',
-    perPage: 10,
-    page: 0,
-    limit: 50,
-    skip: 0,
-    expanded: [],
-  };
+  onSortChange = (property) => (event) => { // eslint-disable-line
+    const { walletHistory } = this.props.store;
 
-  componentWillMount() {
-    this.getTransactions();
+    let order = SortBy.Descending.toLowerCase();
+    if (walletHistory.orderBy === property && walletHistory.direction === SortBy.Descending.toLowerCase()) {
+      order = SortBy.Ascending.toLowerCase();
+    }
+
+    walletHistory.onSortingChange(property, order);
   }
 
-  componentWillReceiveProps(nextProps) {
-    const {
-      transactions,
-      txReturn,
-      syncBlockNum,
-    } = this.props;
+  onRowClick = (id) => (event) => { // eslint-disable-line
+    const { expanded, onExpandedChange } = this.props.store.walletHistory;
 
-    if ((txReturn && !nextProps.txReturn) || (syncBlockNum !== nextProps.syncBlockNum)) {
-      this.getTransactions();
+    const expandedIndex = expanded.indexOf(id);
+    let newExpanded = [];
+    if (expandedIndex === -1) {
+      newExpanded = [...expanded, id];
+    } else {
+      newExpanded = [...expanded.slice(0, expandedIndex), ...expanded.slice(expandedIndex + 1)];
     }
 
-    if (transactions || nextProps.transactions) {
-      const sorted = _.orderBy(
-        nextProps.transactions ? nextProps.transactions : transactions,
-        [this.state.orderBy],
-        [this.state.order],
-      );
+    onExpandedChange(newExpanded);
+  };
 
-      this.setState({
-        transactions: sorted,
-      });
-    }
+  componentDidMount() {
+    this.props.store.walletHistory.queryTransactions();
   }
 
-  componentDidUpdate(prevProps, prevState) {
-    const { skip } = this.state;
+  componentWillReceiveProps(nextProps) {
```

Comment:
is it cool if we just leave this for now? Henry is doing the `txReturn` -> MobX change which directly affects this. Also once we put `syncBlockNum` in MobX this will change as well. We can change this into a reaction in the store instead.

Reply:
yeah that's fine! I will say the implementation is just
```js
componentDidUpdate(prevProps) {
    const { txReturn, syncBlockNum, store: { walletHistory: { queryTransactions } } } = prevProps;

    if ((txReturn && !this.props.txReturn) || (syncBlockNum !== this.props.syncBlockNum)) {
      queryTransactions();
    }
}
```

Reply:
oh ok ill change it if its that easy of a change